File: (8 kyu) Jenny's secret message/(8 kyu) Jenny's secret message.js
// #1
// function greet(name) {
//   if (name === 'Johnny') return 'Hello, my love!';
//   return 'Hello, ' + name + '!';
// }
// #2
// function greet(name) {
//   return name === 'Johnny' ? 'Hello, my love!' : 'Hello, ' + name + '!';
// }
// #3
const greet = (name) => `Hello, ${name === 'Johnny' ? 'my love' : name}!`;

File: (8 kyu) Jenny's secret message/(8 kyu) Jenny's secret message.java
// #1
// public class Greeter {
//   public static String greet(String name) {
//     if(name.equals("Johnny")){
//       return "Hello, my love!";
//     }
//     return String.format("Hello, %s!", name);
//   }
// }

// #2
// public class Greeter {
//   public static String greet(String name) {
//     return (name.equals("Johnny"))
//       ? "Hello, my love!"
//       : String.format("Hello, %s!", name);
//   }
// }

// #3
public class Greeter {
  public static String greet(String name) {
    return String.format("Hello, %s!", (name.equals("Johnny"))
                         ? "my love"
                         : name
                         );
  }
}
